src/pybind/mgr/cephadm/module.py

```diff
@@ -1789,6 +1949,17 @@ def _create_daemon(self,
                     daemon_spec.name(), OrchestratorEvent.ERROR, f'Failed to {what}: {err}')
             return msg
 
+    def _deploy_cephadm_binary(self, host: str) -> bool:
+        # Use tee (from coreutils) to create a copy of cephadm on the target machine
+        self.log.info(f"Deploying cephadm binary to {host}")
+        with self._remote_connection(host) as tpl:
+            conn, _connr = tpl
+            _out, _err, code = remoto.process.check(
+                conn,
+                ['tee', '-', '/var/lib/ceph/{}/cephadm'.format(self._cluster_fsid)],
+                stdin=self._cephadm.encode('utf-8'))
+        return code == 0
```

Comment:
conflicts with #37495

Reply:
Agree - happy to replace once that PR is merged - like this PR it's been open for a while.